mac

```diff
@@ -141,6 +141,21 @@ else
   fancy_echo "Homebrew already installed. Skipping ..."
 fi
 
+# If Xcode command line tools aren't installed, the Homebrew installer will try
+# to install them. But if Homebrew was already installed, then its installer
+# won't run again and we could still be left without Xcode.
+if ! xcode-select -p > /dev/null ; then
+  # From Homebrew's code: https://github.com/Homebrew/install/blob/878b5a18b89ff73f2f221392ecaabd03c1e69c3f/install#L297
+  fancy_echo "Installing Xcode Command Line Tools..."
+  # This temporary file prompts the 'softwareupdate' utility to list the Command Line Tools
+  clt_placeholder="/tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress"
+  sudo /usr/bin/touch "$clt_placeholder"
+  clt_label=$(/usr/sbin/softwareupdate -l | grep -B 1 -E "Command Line (Developer|Tools)" | awk -F"*" '/^ +\\*/ {print $2}' | sed 's/^ *//' | tail -n1 | xargs)
+  sudo /usr/sbin/softwareupdate -i "$clt_label"
+  sudo /bin/rm -f  "$clt_placeholder"
+  sudo /usr/bin/xcode-select --reset
```

Comment:
The Homebrew version uses `--switch` to point to a particular install location `/Library/Developer/CommandLineTools`, but that fails sometimes for some reason.  Using `--reset` tells the system to use its standard search to find the CLT.  I think it works?